### ISSUE TITLE
[Bug] Unable to create MariadbOperator after installing Operator in all namespaces

### ISSUE THREAD

**Issue by philosapiens** [mariadb-operator/mariadb-operator#146]
**Describe the bug**

Once I installed the operator via the OperatorHub into all namespaces and want to create a MariadbOperator instance in a specific namespace, I get the following error message:

```yaml
      message: >-
        failed to install release: rendered manifests contain a resource that
        already exists. Unable to continue with install: ClusterRole
        "mariadb-manager-role" in namespace "" exists and cannot be imported
        into the current release: invalid ownership metadata; label validation
        error: missing key "app.kubernetes.io/managed-by": must be set to
        "Helm"; annotation validation error: missing key
        "meta.helm.sh/release-name": must be set to "{MariadbOperator-name}";
        annotation validation error: missing key
        "meta.helm.sh/release-namespace": must be set to "{current-namespace}"
      reason: InstallError
      status: 'True'
      type: ReleaseFailed
```

**Expected behaviour**

I expect a successful installation. 

In case the operator does not support an installation with installation mode “All namespaces”, it should not be possible to do so. Other operators such as the [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator) disable an installation into all namespaces.

**Steps to reproduce the bug**
 

1. Install Operator into all namespaces
2. Create MariadbOperator instances using the default values

**Environment details**:
- Openshift version: 4.10.59
- Kubernetes version: v1.23.17+16bcd69
- mariadb-operator version: v0.15.1
- Install method: helm, OLM or static manifests
- Install flavour: minimal, recommended or custom

**Comment by mmontes11** [comment]
Hey there @philosapiens ! Thanks for reporting 🙏🏻 

This has to do with the OLM installation, so the next time use [mariadb-operator-helm ](https://github.com/mariadb-operator/mariadb-operator-helm) instead.

You are totally right, `mariadb-manager-role` is not templated and therefore it cannot be installed twice as it is a non-namespaced resource. This is a current limitation we have, it could potentially be solved in the future if the community brings up use cases where having multiple instances of `mariadb-operator` is valuable.  For the time being. I will disable the `AllNamespaces` and the `MultiNamespace` installation modes in the [ClusterServiceVersion](https://github.com/mariadb-operator/mariadb-operator-helm/blob/ed5ce647f0ebc3c2406ff08d8b378dc516d8ccb4/bundle/manifests/mariadb-operator.clusterserviceversion.yaml#L561).

Some users brought up in our [Slack](https://join.slack.com/t/mariadb-operator/shared_invite/zt-1xsfguxlf-dhtV6zk0HwlAh_U2iYfUxw) channel the need of having a single `MariaDB` instance in a single namepsace and specify cross-namespace references in the CRDs:
- https://github.com/mariadb-operator/mariadb-operator/issues/136

Would this be helpful for your use case? If not, could you elaborate a bit more to understand what would work for you?

Thanks 🙏🏻

**Comment by mmontes11** [comment]
> For the time being. I will disable the AllNamespaces and the MultiNamespace installation modes in the [ClusterServiceVersion](https://github.com/mariadb-operator/mariadb-operator-helm/blob/ed5ce647f0ebc3c2406ff08d8b378dc516d8ccb4/bundle/manifests/mariadb-operator.clusterserviceversion.yaml#L561).

Actually, this may not be the way to go. The `installModes` indicate which namespaces the operator can watch, not in which namespaces the operator can be deployed:
- https://sdk.operatorframework.io/docs/building-operators/golang/operator-scope/#updating-your-csvs-installmodes

`mariadb-operator` supports watching resources in any namespace, so having all the [installModes](https://github.com/mariadb-operator/mariadb-operator-helm/blob/ed5ce647f0ebc3c2406ff08d8b378dc516d8ccb4/bundle/manifests/mariadb-operator.clusterserviceversion.yaml#L561...L569) enabled seems correct.

However, the `mariadb-manager-role` `ClusterRoleBinding` is not templated and since it is cluster-wide, you basically cannot have 2 instances of the `mariadb-operator` installed, so solution will be templating this resources in the helm chart. The challenge is that we are currently auto-generating the RBAC resources from Kustmize, so we may have to introduce some sort of advancing templating here:

https://github.com/mariadb-operator/mariadb-operator/blob/2a11b391f4e3480fcae21e3cda48151ae7936f73/make/deploy.mk#L102

Thoughts?

**Comment by mmontes11** [comment]
Hey there @philosapiens !

Installing multiple releases of `mariadb-operator` has been supported in the following PR:
- https://github.com/mariadb-operator/mariadb-operator/pull/151

This includes templating the `mariadb-manager-role` `ClusterRole`, which is your case was clashing with an already existing one.

It will be released in `v0.0.16`! Stay tuned! 🚀
